fabric-rendering-v1/src/main/java/net/fabricmc/fabric/impl/client/rendering/BuiltinItemRendererRegistryImpl.java

```diff
@@ -0,0 +1,31 @@
+package net.fabricmc.fabric.impl.client.rendering;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.item.Item;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.client.rendering.v1.BuiltinItemRenderer;
+import net.fabricmc.fabric.api.client.rendering.v1.BuiltinItemRendererRegistry;
+
+@Environment(EnvType.CLIENT)
+public enum BuiltinItemRendererRegistryImpl implements BuiltinItemRendererRegistry {
+	INSTANCE;
+
+	private static final Map<Item, BuiltinItemRenderer> RENDERERS = new HashMap<>();
+
+	@Override
+	public void register(Item item, BuiltinItemRenderer renderer) {
+		if (RENDERERS.containsKey(item)) {
+			throw new IllegalArgumentException("Item " + item + " already has a builtin renderer!");
+		}
+
+		RENDERERS.put(item, renderer);
+	}
+
+	public static BuiltinItemRenderer getRenderer(Item item) {
```

Comment:
We don't have annotations yet for nullability we agree on, so a simple `/* NULLABLE */` comment would be nice for when we go back to manage these